docs/examples/MondrianTreeRegressor.md

```diff
@@ -0,0 +1,273 @@
+from skgarden import MondrianTreeRegressor
+from itertools import cycle
+%matplotlib inline
+```
+
+##  Extremely Randomized Tree - Recap
+
+In a tree-based algorithm, the decision rule at every node is constructed by considering a set of candidate split points and that candidate split point that minimizes a certain impurity (generally the mse in case of continuous values) is chosen. In an extremely randomized tree, the number of candidate splits `S` are limited to `n_features` and each candidate split `S[i]` is drawn unifomly from the bounds `(l_f[i], u_f[i])`.
+
+In a single extremely randomized tree, this construction might seem suboptimal but while constructing an ensemble of trees, it makes sure that each tree constructed in an independent fashion. In addition, limiting the number of candidate split points makes the tree construction very fast. Let us now generate some toy data.
+
+
+```python
+def generate_toy_data(n_points=10):
+    X_train = np.linspace(-np.pi, np.pi, n_points)
+    y_train = np.sin(X_train)
+    X_test = np.linspace(-6.0, 6.0, 100)
+    y_test = np.sin(X_test)
+    return np.expand_dims(X_train, axis=1), y_train, np.expand_dims(X_test, axis=1), y_test
+
+X_train, y_train, X_test, y_test = generate_toy_data()
+plt.plot(X_train.ravel(), y_train, "ro")
+plt.show()
+```
+
+
+![png](../mondrian_tree/plot1.png)
+
+
+## Plotting decision boundaries using ERT's
+
+Let us now scikit-learn's ExtraTreeRegressor to train on the generated toy data, predict on some unseen data and plot decision boundaries in the 1-D space. Also, we set the `max_depth` parameter to 2, which means there can be a maximum of 4 decision boundaries in the 1-D space.
+
+
+```python
+def plot_1D_decision_boundaries(tree_reg, plt):
+    x_l, x_u = plt.xlim()
+    colors = cycle("bgrcmyk")
+    split_thresh = tree_reg.tree_.threshold
+    split_thresh = sorted(split_thresh[split_thresh != -2])
+    split_thresh = [x_l] + split_thresh + [x_u]
+    for s in split_thresh:
+        plt.axvline(s, color="r", linestyle="-")
+
+    for x_l, x_u in zip(split_thresh[1:], split_thresh[:-1]):
+        plt.fill_between((x_l, x_u), -1.5, 1.5, alpha=0.2, color=next(colors))
+
+etr = ExtraTreeRegressor(random_state=2, max_depth=2)
+etr.fit(X_train, y_train)
+y_pred = etr.predict(X_test)
+plt.plot(X_train.ravel(), y_train, "ro")
+plt.plot(X_test.ravel(), y_pred)
+plt.ylim((-1.5, 1.5))
+plot_1D_decision_boundaries(etr, plt)
+plt.show()
+```
+
+![png](../mondrian_tree/plot2.png)
+
+The blue line represents the mean prediction in every region of the decision space.
+
+We notice that as we move away from the training data
+
+1. The predicted mean remains constant.
+2. It is determined by the decision split at the bounds of the space.
+
+Ideally, as move away from the training data we are unsure about the target value to predict and would like to fall back on some prior mean. The mondrian tree solves this problem in a very intelligent way.
+
+##  Mondrian Tree
+
+### Train time
```

Comment:
I would rename this section -- it feels like it concerns timing issues when building a tree :)